src/core/calculate_pf.jl

```diff
@@ -0,0 +1,601 @@
+    )
+
+Computes native power flow and outputs the result dict.
+"""
+function compute_pf(data_math::Dict{String, Any}; v_start::Union{Dict{<:Any,<:Any},Missing}=missing, max_iter::Int=100, stat_tol::Float64=1E-8, verbose::Bool=true)
+    if !ismultinetwork(data_math)
+        nw_dm = Dict("0"=>data_math)
+    else
+        nw_dm = data_math["nw"]
+    end
+
+    sol = Dict{String, Any}()
+    time_build = Dict{String, Any}()
+    time_solve = Dict{String, Any}()
+    time_post = Dict{String, Any}()
+    Yf_matrix = Dict{String, Any}()
+    Yv_matrix = Dict{String, Any}()
+    status = Dict{String, Any}()
+    its = Dict{String, Any}()
+    stat = Dict{String, Any}()
+
+    for (nw, dm) in nw_dm
+        if ismissing(v_start)
+            add_start_vrvi!(dm)
+            v_start = _bts_to_start_voltage(dm)
+        end
+
+
+        time_build[nw] = @elapsed pfd = PowerFlowData(dm, v_start)
+
+        time_solve[nw] = @elapsed (Uv, status[nw], its[nw], stat[nw]) = _compute_Uv(pfd, max_iter=max_iter, stat_tol=stat_tol)
+
+        time_post[nw] = @elapsed sol[nw] = build_solution(pfd, Uv)
+
+        Yf_matrix[nw] = pfd.Yf
+        Yv_matrix[nw] = pfd.Yv_LU
+    end
+
+    res = Dict{String, Any}()
+    if !ismultinetwork(data_math)
+        res["time_build"] = time_build["0"]
+        res["time_solve"] = time_solve["0"]
+        res["time_post"] = time_post["0"]
+        res["solution"] = sol["0"]
+        res["termination_status"] = status["0"]
+        res["iterations"] = its["0"]
+        res["stationarity"] = stat["0"]
+        res["time_total"] = time_build["0"]+time_solve["0"]+time_post["0"]
+        res["Yf"] = Yf_matrix["0"]
+        res["Yv"] = Yv_matrix["0"]
+    else
+        res["time_build"] = time_build
+        res["time_solve"] = time_solve
+        res["time_post"] = time_post
+        res["solution"] = sol
+        res["termination_status"] = status
+        res["iterations"] = its
+        res["stationarity"] = stat
+        res["time_total"] = sum(values(time_build))+sum(values(time_solve))+sum(values(time_post))
+        res["all_converged"] = all(x==CONVERGED for (_, x) in status)
+        res["Yf"] = Yf_matrix
+        res["Yv"] = Yv_matrix
+    end
+
+    return res
+end
+
+"""
+    compute_pf(
+      pdf::PowerFlowData,
+      max_iter::Int,
+      stat_tol::Float,
+      verbose::Bool
+    )
+
+Constructor for compute_pf that requires PowerFlowData.
+"""
+function compute_pf(pfd::PowerFlowData; max_iter::Int=100, stat_tol::Float64=1E-8, verbose::Bool=true)
+    time = @elapsed (Uv, status, its, stat) = _compute_Uv(pfd, max_iter=max_iter, stat_tol=stat_tol)
+    return build_result(pfd, Uv, status, its, time, stat)
+end
+
+"""
+    _compute_Uv(
+      pfd::PowerFlowData,
+      max_iter::Int,
+      stat_tol::Float,
+      verbose::Bool
+    )
+
+Computes variable voltages until convergence or reaching max_iter.
+"""
+function _compute_Uv(pfd::PowerFlowData; max_iter::Int=100, stat_tol::Float64=1E-8, verbose::Bool=true)
```

Comment:
this docstring needs to mention that 
- this is the fixed-point current injection method in the style of OpenDSS.
- conversely to OpenDSS, it does not switch load models depending on the voltage

we should also add a page to the docs that describes the core algorithm functionality. 
however, we may want to keep this as an item for discussion in the PR to PMD proper, and first ask if they prefer a separate pdf report, a small note in the docs, or an extended description in the docs.